sourcespec/ssp_build_spectra.py

```diff
@@ -179,23 +203,27 @@ def _cut_signal_noise(config, trace):
                          endtime=noise_t1+config.spectral_win_length,
                          pad=True,
                          fill_value=0)
-    # Be sure that both traces have same length:
-    if len(trace_signal) != len(trace_noise):
-        npts = min(len(trace_signal), len(trace_noise))
-        trace_signal.data = trace_signal.data[:npts]
-        trace_noise.data = trace_noise.data[:npts]
 
     return trace_signal, trace_noise
 
 
-def _check_noise_level(trace_signal, trace_noise):
+def _check_noise_level(trace_signal, trace_noise, config):
     traceId = trace_signal.get_id()
     trace_signal_rms = ((trace_signal.data**2).sum())**0.5
-    trace_noise_rms = ((trace_noise.data**2).sum())**0.5
-    if trace_noise_rms/trace_signal_rms < 1e-6:
+    # Scale trace_noise_rms to length of signal window,
+    # based on length of non-zero noise window
+    try:
+        scale_factor = float(len(trace_signal)) / len(trace_noise.data != 0)
+    except ZeroDivisionError:
+        scale_factor = 1
+    trace_noise_rms = ((trace_noise.data**2 * scale_factor).sum())**0.5
+    if trace_noise_rms/trace_signal_rms < 1e-6 and config.weighting == 'noise':
         msg =\
-            '{}: Noise level is too low or zero: ignoring for noise weighting'
+            '{}: Noise level is too low or zero: station will be skipped'
         msg = msg.format(traceId)
+        # I think this error should only be raised if noise weighting is used
```

Comment:
Can you move this comment below line 220 and say something like:
```
# Skip trace only if noise weighting is used
```

Reply:
Done.